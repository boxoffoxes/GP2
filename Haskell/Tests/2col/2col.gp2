Main = init; { prop_red, prop_blue }!

init (x:list) [
	(n1, x)
| ] => [
	(n1, x # red)
|]
interface = {n1}

prop_red (x,y,z:list) [
	(n1, x # red)
	(n2, y)
|
	(e1 (B), n1, n2, z)
] => [
	(n1, x # red)
	(n2, y # blue)
|
	(e1 (B), n1, n2, z)
]
interface = {n1,n2}

prop_blue (x,y,z:list) [
	(n1, x # blue)
	(n2, y)
|
	(e1 (B), n1, n2, z)
] => [
	(n1, x # blue)
	(n2, y # red)
|
	(e1 (B), n1, n2, z)
]
interface = {n1,n2}

