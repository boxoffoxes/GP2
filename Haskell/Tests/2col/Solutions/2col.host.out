[
 (n1, empty # red)
 (n2, empty # blue)
 (n3, empty # red)
 (n4, empty # blue)
 (n5, empty # red)
 (n6, empty # blue)
 (n7, empty # red)
 (n8, empty # blue)
 (n9, empty # red)
 (n10, empty # blue)
 (n11, empty # red)
 (n12, empty # blue)
 (n13, empty # red)
 (n14, empty # blue)
 (n15, empty # red)
 (n16, empty # blue)
 (n17, empty # red)
 (n18, empty # blue)
 (n19, empty # red)
 (n20, empty # blue)
 (n21, empty # red)
 (n22, empty # blue)
 (n23, empty # red)
 (n24, empty # blue)
 (n25 (R), empty # red)
|
 (e21, n2, n1, empty)
 (e22, n3, n2, empty)
 (e23, n4, n3, empty)
 (e24, n5, n4, empty)
 (e1, n6, n1, empty)
 (e2, n7, n2, empty)
 (e25, n7, n6, empty)
 (e3, n8, n3, empty)
 (e26, n8, n7, empty)
 (e4, n9, n4, empty)
 (e27, n9, n8, empty)
 (e5, n10, n5, empty)
 (e28, n10, n9, empty)
 (e6, n11, n6, empty)
 (e7, n12, n7, empty)
 (e29, n12, n11, empty)
 (e8, n13, n8, empty)
 (e30, n13, n12, empty)
 (e9, n14, n9, empty)
 (e31, n14, n13, empty)
 (e10, n15, n10, empty)
 (e32, n15, n14, empty)
 (e11, n16, n11, empty)
 (e12, n17, n12, empty)
 (e33, n17, n16, empty)
 (e13, n18, n13, empty)
 (e34, n18, n17, empty)
 (e14, n19, n14, empty)
 (e35, n19, n18, empty)
 (e15, n20, n15, empty)
 (e36, n20, n19, empty)
 (e16, n21, n16, empty)
 (e17, n22, n17, empty)
 (e37, n22, n21, empty)
 (e18, n23, n18, empty)
 (e38, n23, n22, empty)
 (e19, n24, n19, empty)
 (e39, n24, n23, empty)
 (e20, n25, n20, empty)
 (e40, n25, n24, empty)
]
